### PR TITLE
Logged hours issue

### PR DESCRIPTION
**Pull request by shivamchouhan123**

### CODE REVIEW

File: app/controllers/merged_requests_controller.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    @merged_request = @task.merged_requests.find(params[:id])
+  end
+
+  def update
+    @merged_request = @merged_request.update(merged_request_params)
+  end
+
+  def destroy
+    @merged_request.destroy
+    redirect_to task_logged_hours_url, notice: 'Hour was successfully destroyed.'
```
Comment (ror-mahesh) [inline app/controllers/merged_requests_controller.rb:33]: notice message wrong.

File: app/views/merged_requests/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      <%= form.label :raised_date %>
+      <%= form.date_select :raised_date, class: 'form-control', selected: @merged_request.raised_date %>
+   </div>
+   <div class="mb-3">
+      <label for="message-text" class="col-form-label">Description:</label>
+      <%= form.text_area :description, class: 'form-control', selected: @merged_request.description %>
+   </div>
+   <div class="mb-3">
+      <%= form.label :status %>
+      <%= form.select :status, [['Open', 'open'], ['Closed', 'closed'], ['Draft', 'draft'], ['Merged', 'merged']], {}, class: 'form-control', selected: @merged_request.status %>
```
Comment (ror-mahesh) [inline app/views/merged_requests/_form.html.erb:17]: can we use enum for status.

File: app/views/merged_requests/_mr.html.erb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            <%= merge_request.raised_date %>
+         </td>
+         <td>
+            <%= merge_request.description %>
+         </td>
+         <td>
+            <%= merge_request.status %>
+         </td>
+          <td>
+            <% if merge_request&.id&.present? %> <%= link_to 'Edit' , edit_task_merged_request_path(@task, merge_request), class: "btn btn-primary mb-3 right", data: { bs_toggle: "modal",bs_target: "#mergedRequestModal",remote: true } %> <% end %>
```
Comment (ror-mahesh) [inline app/views/merged_requests/_mr.html.erb:26]: no need to use `&` here also why we check MR id here?.

File: app/views/merged_requests/create.js.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<% if @merged_request.errors.any? %>
+	alert("Invalid details");
```
Comment (ror-mahesh) [inline app/views/merged_requests/create.js.erb:2]: can we show actual error instead of static message.

**Review by ror-mahesh — CHANGES_REQUESTED** [review]

File: app/controllers/merged_requests_controller.rb (hunk 1 above)
Comment (shivamchouhan123) [reply app/controllers/merged_requests_controller.rb:33]: so here can i use Merged Request successfully destroyed.

File: app/models/merged_request.rb [hunk 5]
```diff
@@ -1,3 +1,15 @@
+# frozen_string_literal: true
+
 class MergedRequest < ApplicationRecord
   belongs_to :task
+   validates :url_link, presence: true
+
```
Comment (ror-mahesh) [inline app/models/merged_request.rb:6]: remove extra line.

File: app/views/merged_requests/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+<%= form_for([@task, @merged_request], remote: true) do |form| %>
+<div class="modal-body">
+   <div class="field mt-3 mb-3">
+      <%= form.label :url_link %>
+      <%= form.text_field :url_link,  value: "#{@merged_request.url_link}", class: "form-control" %>
```
Comment (ror-mahesh) [inline app/views/merged_requests/_form.html.erb:5]: why we pass value here,
Rails form already load the value of attributes.

**Review by ror-mahesh — CHANGES_REQUESTED** [review]

File: app/views/tasks/_mr.html.erb [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<div id="task">
+   <%= form_for([@task, @merged_request], remote: true) do |form| %>
+   <div class="modal-body">
+      <div class="field mt-3 mb-3">
+         <%= form.label :url_link %>
+         <%= form.time_field :url_link,  value: "#{@logged_hour.total_time}", class: "form-control" %>
```
Comment (ror-mahesh) [inline app/views/tasks/_mr.html.erb:6]: here also value passed?

**Review by ror-mahesh — CHANGES_REQUESTED** [review]

**Review by ror-mahesh — APPROVED** [review]

**Pull request merged**